src/main/kotlin/org/rust/lang/core/type/RustResolvedType.kt

```diff
@@ -0,0 +1,26 @@
+package org.rust.lang.core.type
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.SyntheticElement
+import org.rust.lang.core.psi.RustImplItem
+import org.rust.lang.core.psi.RustImplMethodMember
+import org.rust.lang.core.psi.impl.mixin.isStatic
+
+
+interface RustResolvedType: PsiElement, SyntheticElement {
```

Comment:
What is the motivation to make types part of the PSI hierarchy? Seems questionable to me.